components/tabs/tabs.js

```diff
@@ -0,0 +1,388 @@
+				left: 0;
+				margin-left: 4px;
+				margin-right: 0;
+				right: auto;
+			}
+			.d2l-tabs-container[allow-scroll-next] > .d2l-tabs-scroll-next-container {
+				display: inline-block;
+			}
+			.d2l-tabs-scroll-button {
+				background-color: transparent;
+				border: 1px solid transparent;
+				border-radius: 15px;
+				box-shadow: 0 0 0 4px rgba(0, 0, 0, 0);
+				box-sizing: border-box;
+				cursor: pointer;
+				display: inline-block;
+				height: 30px;
+				margin: 0;
+				outline: none;
+				padding: 0;
+				transform: translateY(10px);
+				-webkit-transition: box-shadow 0.2s;
+				transition: box-shadow 0.2s;
+				width: 30px;
+			}
+			.d2l-tabs-scroll-button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+			.d2l-tabs-scroll-button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-tabs-scroll-button[disabled]:hover,
+			.d2l-tabs-scroll-button[disabled]:focus {
+				background-color: transparent;
+			}
+			.d2l-tabs-scroll-button:hover,
+			.d2l-tabs-scroll-button:focus {
+				background-color: var(--d2l-color-gypsum);
+			}
+			.d2l-tabs-scroll-button:focus {
+				border-color: var(--d2l-color-celestine);
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+
+			}
+		`];
+	}
+
+	static get resources() {
+		return {
+			'ar': { 'scroll.previous': 'التمرير إلى اليسار', 'scroll.next':'التمرير إلى اليمين' },
+			'en': { 'scroll.previous': 'Scroll Backward', 'scroll.next': 'Scroll Forward' },
+			'es': { 'scroll.previous': 'Desplácese a la izquierda', 'scroll.next': 'Desplácese a la derecha' },
+			'fr': { 'scroll.previous': 'Défilement vers la gauche', 'scroll.next': 'Défilement vers la droite' },
+			'ja': { 'scroll.previous': '左にスクロール', 'scroll.next': '右にスクロール' },
+			'ko': { 'scroll.previous': '왼쪽으로 스크롤', 'scroll.next': '오른쪽으로 스크롤' },
+			'nl': { 'scroll.previous': 'Naar links schuiven', 'scroll.next': 'Naar rechts schuiven' },
+			'pt': { 'scroll.previous': 'Rolar para Esquerda', 'scroll.next': 'Rolar para Direita' },
+			'sv': { 'scroll.previous': 'Rulla till vänster', 'scroll.next': 'Rulla till höger' },
+			'tr': { 'scroll.previous': 'Sola Kaydır', 'scroll.next': 'Sağa Kaydır' },
+			'zh': { 'scroll.previous': '向左滚动', 'scroll.next': '向右滚动' },
+			'zh-tw': { 'scroll.previous': '向左捲動', 'scroll.next': '向右捲動' }
+		};
+	}
+
+	constructor() {
+		super();
+		this.maxToShow = -1;
+		this._tabInfos = [];
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		queueMicrotask(() => {
+			const bgColor = this._getComputedBackgroundColor();
+			if (bgColor && bgColor !== 'rgb(255, 255, 255)' && bgColor !== 'rgba(255, 255, 255, 1)') {
+				this.style.setProperty('--d2l-tabs-background-color', bgColor);
+			}
+		});
+
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this.arrowKeysFocusablesProvider = async() => {
+			return [...this.shadowRoot.querySelectorAll('d2l-tab')];
+		};
+		/*
+		this.arrowKeysOnBeforeFocus = async(tab) => {
```

Comment:
This will be updated in follow-up commits.  Basically, we use this callback to update the virtual scroll position before focus is applied to prevent the browser from (automatically) natively scrolling the focused element into view.